torch_frame/nn/encoder/stype_encoder.py

```diff
@@ -590,3 +594,71 @@ def encode_forward(self, feat: Tensor) -> Tensor:
         # -> [batch_size, num_cols, out_channels]
         x = x_lin + self.bias
         return x
+
+
+class LinearEmbeddingModelEncoder(StypeEncoder):
+    r"""Linear function based encoder with a specified model outputs embedding
+    features. It applies a linear layer :obj:`torch.nn.Linear(in_channels,
+    out_channels)` on each embedding feature (:obj:`in_channels` is the
+    dimensionality of the embedding) and concatenates the output embeddings.
+    Note that the implementation does this for all :obj:`text_tokenized`
+    features in a batched manner.
+
+    Args:
+        in_channels (int): The dimensionality of the embedding feature. Needs
+            to be specified manually.
+        model (torch.nn.Module): The model outputs embedding features that will
+            be input into the linear layer.
+    """
+    # NOTE: We currently support text embeddings but in principle, this encoder
+    # can support any model outputs embeddings, including image/audio/graph
+    # embeddings.
+    supported_stypes = {stype.text_tokenized}
+
+    def __init__(
+        self,
+        out_channels: Optional[int] = None,
+        stats_list: Optional[List[Dict[StatType, Any]]] = None,
+        stype: Optional[stype] = None,
+        post_module: Optional[Module] = None,
+        na_strategy: Optional[NAStrategy] = None,
+        in_channels: Optional[int] = None,
+        model: Optional[torch.nn.Module] = None,
+    ):
+        if in_channels is None:
+            raise ValueError("Please manually specify the `in_channels`, "
+                             "which is the text embedding dimensionality.")
```

Comment:
```suggestion
                             "specifying the text embedding dimensionality.")
```
Why do we make these arguments optional if they're not, in fact, optional? Can't we just drop these errors and make these args positional?

Reply:
To make them not optional we need to move it before the arguments with default values. But then it will be conflict with `Module` initialization and thus cause some error during `validate` cc @weihua916 We may need to improve this in the future.